api/src/task/notifications/generate_notifications.py

```diff
@@ -150,9 +173,30 @@ def _send_notifications(self) -> None:
             if not container.saved_opportunities and not container.saved_searches:
                 continue
 
-            # TODO: Implement actual notification sending in future ticket
+            # Get user email from LinkExternalUser using select pattern
+            stmt = select(LinkExternalUser).where(LinkExternalUser.user_id == user_id)
+            user = self.db_session.execute(stmt).scalar_one_or_none()
+
+            if not user or not user.email:
+                logger.warning("No email found for user", extra={"user_id": user_id})
+                continue
+
+            # Send email via Pinpoint
+            subject = "Updates to Your Saved Opportunities"
+            message = (
+                f"You have updates to {len(container.saved_opportunities)} saved opportunities"
+            )
+
+            send_pinpoint_email_raw(
+                to_address=user.email,
+                subject=subject,
+                message=message,
+                pinpoint_client=self.pinpoint_client,
+                app_id=self.config.app_id,
+            )
```

Comment:
Probably want a try/except around this - if it fails, we should update the `UserNotificationLog` notification_sent field.